libraries/Bluefruit52Lib/src/clients/BLEClientHidAdafruit.cpp

```diff
@@ -195,3 +209,33 @@ void BLEClientHidAdafruit::getMouseReport(hid_mouse_report_t* report)
   memcpy(report, &_last_mse_report, sizeof(hid_mouse_report_t));
 }
 
+/*------------------------------------------------------------------*/
+/* Gamepad
+ *------------------------------------------------------------------*/
+bool BLEClientHidAdafruit::gamepadPresent(void)
+{
+  return _gpd_boot_input.discovered();
```

Comment:
gamepad detection can only be done by parsing the HID Report descriptors, and also use it to determine the gamepad report layout (PS4, XBox one etc..) This is probably too much for nRF52 client, that is why in the central_hid example, I immediately switch to Boot mode so that it could work with consumer keyboard on the market
https://github.com/adafruit/Adafruit_nRF52_Arduino/blob/master/libraries/Bluefruit52Lib/examples/Central/central_hid/central_hid.ino#L141

For Gamepad, a full parser is probably too much, we can actually need to do some hack at sketch-level e.g use the Device Information Service, manufacturer to write a specific sketch e.g "Use BLE PS4 Controller to control remote car" that the sketch know exactly how to interpreter report. The host/central side is always is more complicated than its prph counterpart. However, you don't really need to worry about this for now. We can leave central side as it is, no problem at all, it currently only communicate through boot endpoint anyway.